commons-core/src/main/scala/com/avsystem/commons/rpc/rpcMetadata.scala

```diff
@@ -0,0 +1,72 @@
+package com.avsystem.commons
+package rpc
+
+import com.avsystem.commons.serialization.{HasGenCodec, transparent}
+
+/**
+  * This trait must be extended by all method metadata classes and all parameter metadata classes.
+  * For method metadata, type parameter `T` will be matched against each real method result type.
+  * For parameter metadata, type parameter `T` will be matched against each real parameter type.
+  */
+trait TypedMetadata[T]
+
+/**
+  * Information about real parameter flags and modifiers as defined in Scala code.
+  */
+@transparent
+final case class ParamFlags(rawFlags: Int) extends AnyVal {
```

Comment:
`Enumeration` + `ValueSet` would be more object friendly and less off-by-one error-prone.

Reply:
Did you just suggest using `scala.Enumeration` to me?

Reply:
I'd rather `Enumeration` than C-style bit-fiddling. ValueSet is actually quite a nice wrapper around BitSet.

Reply:
I refuse to use `Enumeration` which uses reflection and is widely disgusted. I'm going to introduce some safety measures to avoid off-by-one errors in refactorings.

Reply:
Conscience clause of Scala developers